### DISCUSSION TITLE
Active Directory Interactive Authentication to connect azure sql db using Microsoft.Data.Client library

### DISCUSSION THREAD

**Discussion by somnath-pramanik123** [microsoft/azuredatastudio#15833] [General]
I replaced System.Data.SqlClient to Microsoft.Data.SqlClient(Version 3.0.0) in MVC Controller class and below is the code for connecting azure sql. I am using Active Directory Interactive Authentication to connect azure sq DB. The target Framework is .NET 4.8`
            
        public async Task<JsonResult> Index(){
       
     //Connection string ===========================
    
            SqlConnection con = new SqlConnection($"Server={serverName}.database.windows.net;UID=Anystring;Authentication=Active Directory Interactive; Database={dbName}; MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=true;Connection Timeout=30;");
        try{
                            using (var con = new SqlConnection(connectionString))
                            {
     //Opening connection ===========     
                         con.Open(SqlConnectionOverrides.OpenWithoutRetry);
        
                                var sql = $"SELECT newid()";
                                using (var cmd = new SqlCommand(sql, con))
                                {
                                    var result = cmd.ExecuteScalar().ToString();
        
                                    var endTime = DateTime.Now;
                                    return Json(new
                                    {                                        
                                    }, JsonRequestBehavior.AllowGet);
                                }
                            }
        }
        catch (Exception ex)
                    {
        return null;
        }
        
        }

When trying to run the code , connection to DB is not successful.
        The same code is running perfectly fine in Console Application without any issue.
         
        Any thought, what could be the root cause of the issue.

**Comment by Charles-Gagnon** [comment]
I'm unsure how this is related to Azure Data Studio - it seems like https://github.com/dotnet/SqlClient might be a better place to ask this question?
